src/internal/components/dropdown/styles.scss

```diff
@@ -50,6 +50,11 @@
     }
 
     border-bottom: none;
+
+    // Don't show dropdown border when there's no dropdown content
+    &.is-empty::after {
```

Comment:
The `::after` pseudo element is being rendered on this node, so this gets the empty styles.